discussions/src/lib.rs

```diff
@@ -0,0 +1,50 @@
+mod social_db;
+
+use crate::social_db::social_db_contract;
+use near_sdk::borsh::{BorshDeserialize, BorshSerialize};
+use near_sdk::{env, near_bindgen, require, AccountId, NearToken, Promise};
+
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize, Default)]
+#[borsh(crate = "near_sdk::borsh")]
+pub struct Contract {}
+
+#[near_bindgen]
+impl Contract {
+    #[init]
+    pub fn new() -> Self {
+        social_db_contract()
+            .with_unused_gas_weight(1)
+            .with_attached_deposit(NearToken::from_near(1))
+            .grant_write_permission(
+                Some(Contract::get_devhub_account()),
+                None,
+                vec![env::current_account_id().to_string()],
+            );
+        Contract {}
+    }
+
+    pub fn destroy(&mut self) {
```

Comment:
Unless there a very good reason not to, all the cross-contract calling functions should return the Promise as the result of its execution, otherwise, the cross-contract calls won't affect the result of the contract execution.

Let me try to explain with a quick example:

```rust
fn destroy(&mut self) {
    Promise::...; // notice the `;` here and the empty return value in the function signature
}
```


```rust
fn destroy(&mut self) -> Promise {
    Promise::... // notice there is no `;` here and the return value in the function signature is Promise
}
```

The difference between the two is that the first will schedule the cross-contract call, but its return value will be ignored (even if the call fails, the result of the `destroy` function will be OK), and transaction return value may even be returned earlier than all the in-flight cross-contract calls complete since they do not affect the execution result.

Without linking the cross-contract call with the result of `destroy` function, the `destroy` function in the community contract may start removing itself earlier than the destruction of the discussion account happens and if discussion account destruction fails, it will reach into unreachable state.

I believe the other issues that you raised above will be resolved once you fix this everywhere where cross-contract results are currently simply ignored and do not contribute to the final result.